### PR TITLE
Various tweaks, read inside

### PR DESCRIPTION
**Pull request by Sunconure11**
1. Fixed up the gradle file a bit, it should properly grab Rustic from curse now.
2. Started on Thaumcraft support. A small example of how to pull it off is in the grapes module (aspects). Baubles had to be installed to the workspace so I could test my Thaumcraft support, since Thaumcraft relies on it.
3. Added JEI and HWYLA to the workspace
4. Various gradle-related tweaks, to aid me in even starting a build in the first place
5. Updated Forge to the current recommended version.
6. Suppressed various deprecation warnings.

Todo:

Figure out how to pull Harvestcraft from curse properly. The versioning is insane. I don't understand how I did it for Rustic and Thaumcraft, and Harvestcraft's many oddities are making it more confusing.

https://authors.curseforge.com/docs/api

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Sets default memory used for gradle commands. Can be overridden by user or command line properties.
 # This is required to provide enough memory for the Minecraft decompilation process.
 org.gradle.jvmargs=-Xmx3G
-mod_version=3.1.0
+mod_version=4.0.1
```
Comment (IceDragon200) [inline gradle.properties]: If your still following my versioning, then 1.12 should have been 7.x

| MC | GC |
| --- | --- |
| 1.7 | 2.x |
| 1.8 | 3.x |
| 1.9 | 4.x |
| 1.10 | 5.x |
| 1.11 | 6.x | 
| 1.12 | 7.x |

Using a different major version for GC for a different version of minecraft means users would get less confused (which GC version works with which version of MC)

Comment (Sunconure11) [reply gradle.properties]: That versioning system hasn't been used for quite a few MC versions.

https://github.com/GrowthcraftCE/Growthcraft-1.12/blob/development/build.properties

https://github.com/GrowthcraftCE/Growthcraft-1.11/blob/development/gradle.properties

https://github.com/GrowthcraftCE/Growthcraft-1.10/blob/development/gradle.properties

Comment (IceDragon200) [reply gradle.properties]: Yeah, looks like they forgot to bump it with each version

**Comment by Avatair** [comment]
Directing the PR to development-sunconure instead, because PR requires more review

**Pull request merged**